File: backend/src/main/java/com/hrms/domain/survey/Survey.java
package com.hrms.domain.survey;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "surveys")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Survey {
    @Id @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    @Column(name = "tenant_id", nullable = false)
    private UUID tenantId;
    @Column(name = "survey_code", nullable = false, length = 50)
    private String surveyCode;
    @Column(name = "title", nullable = false, length = 200)
    private String title;
    @Column(name = "description", columnDefinition = "TEXT")
    private String description;
    @Enumerated(EnumType.STRING)
    @Column(name = "survey_type", length = 50)
    private SurveyType surveyType;
    @Column(name = "is_anonymous")
    private Boolean isAnonymous = false;
    @Column(name = "start_date")
    private LocalDateTime startDate;
    @Column(name = "end_date")
    private LocalDateTime endDate;
    @Enumerated(EnumType.STRING)
    @Column(name = "status", length = 20)
    private SurveyStatus status = SurveyStatus.DRAFT;
    @Column(name = "target_audience", length = 50)
    private String targetAudience;
    @Column(name = "total_responses")
    private Integer totalResponses = 0;
    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    @Column(name = "created_by")
    private UUID createdBy;
    @Version
    private Long version = 0L;

    public enum SurveyType {
        ENGAGEMENT, SATISFACTION, PULSE, EXIT, FEEDBACK, CUSTOM
    }

    public enum SurveyStatus {
        DRAFT, ACTIVE, PAUSED, COMPLETED, ARCHIVED
    }
}

File: frontend/lib/types/grow/survey.ts
// Survey Types

export enum SurveyType {
  ENGAGEMENT = 'ENGAGEMENT',
  SATISFACTION = 'SATISFACTION',
  PULSE = 'PULSE',
  EXIT = 'EXIT',
  FEEDBACK = 'FEEDBACK',
  CUSTOM = 'CUSTOM',
}

export enum SurveyStatus {
  DRAFT = 'DRAFT',
  ACTIVE = 'ACTIVE',
  PAUSED = 'PAUSED',
  COMPLETED = 'COMPLETED',
  ARCHIVED = 'ARCHIVED',
}

export enum QuestionType {
  SINGLE_CHOICE = 'SINGLE_CHOICE',
  MULTIPLE_CHOICE = 'MULTIPLE_CHOICE',
  TEXT = 'TEXT',
  RATING = 'RATING',
  SCALE = 'SCALE',
  YES_NO = 'YES_NO',
  DATE = 'DATE',
  NPS = 'NPS',
}

export interface Survey {
  id: string;
  surveyCode: string;
  title: string;
  description?: string;
  surveyType: SurveyType;
  isAnonymous: boolean;
  startDate?: string;
  endDate?: string;
  status: SurveyStatus;
  targetAudience?: string;
  totalResponses: number;
  createdAt: string;
  updatedAt: string;
  createdBy?: string;
  questions?: SurveyQuestion[];
}

export interface SurveyRequest {
  surveyCode: string;
  title: string;
  description?: string;
  surveyType: SurveyType;
  isAnonymous?: boolean;
  startDate?: string;
  endDate?: string;
  status?: SurveyStatus;
  targetAudience?: string;
}

export interface SurveyQuestion {
  id: string;
  surveyId: string;
  questionText: string;
  questionType: QuestionType;
  isRequired: boolean;
  orderIndex: number;
  options?: string[];
  minValue?: number;
  maxValue?: number;
}

export interface QuestionRequest {
  surveyId: string;
  questionText: string;
  questionType: QuestionType;
  isRequired?: boolean;
  orderIndex?: number;
  options?: string[];
  minValue?: number;
  maxValue?: number;
}

export interface SurveyResponse {
  id: string;
  surveyId: string;
  employeeId?: string;
  submittedAt: string;
  answers: SurveyAnswer[];
}

export interface SurveyAnswer {
  id: string;
  questionId: string;
  answerText?: string;
  selectedOptions?: string[];
  ratingValue?: number;
}

export interface SubmitResponseRequest {
  surveyId: string;
  employeeId?: string;
  answers: {
    questionId: string;
    answerText?: string;
    selectedOptions?: string[];
    ratingValue?: number;
  }[];
}

export interface SurveyAnalyticsSummary {
  surveyId: string;
  totalResponses: number;
  completionRate: number;
  averageCompletionTime?: number;
  questionStats: QuestionStat[];
}

export interface QuestionStat {
  questionId: string;
  questionText: string;
  responseCount: number;
  averageRating?: number;
  optionDistribution?: Record<string, number>;
}

export interface EngagementScore {
  id: string;
  employeeId: string;
  surveyId: string;
  score: number;
  calculatedAt: string;
  category: string;
}
